### PR TITLE
Move core_validation to layer chassis

### PR DESCRIPTION
**Pull request by mark-lunarg**
This PR should get VVL to an all-chassis layer configuration, but there is plenty of follow-on work to get to a final state.  For instance, several hoops were jumped through to make the use of `device_data `and `instance_data `(along with `GetLayerDataPtr `and `get_dispatch_key`) benign within the CoreChecks validation object.  This will all go away in the following week or two.  The goal was to have the minimum of source-code changes across the conversion, though it's still a big change -- too big to not have resulting problems.  Seems OK on Android, Linux, Windows, internal CI, and Dota2, but I'll have to address the hidden bugs as they come up.

The main changeover happens in this commit:  `layers: Convert CV to use layer chassis`

### CODE REVIEW

**Comment by mark-lunarg** [comment]
Had to rebase on Tony's change.

File: scripts/layer_chassis_generator.py [hunk 1]
```diff
@@ -254,6 +256,53 @@ class LAYER_PHYS_DEV_PROPERTIES {
     std::vector<VkQueueFamilyProperties> queue_family_properties;
 };
 
+// CHECK_DISABLED struct is a container for bools that can block validation checks from being performed.
+// The end goal is to have all checks guarded by a bool. The bools are all "false" by default meaning that all checks
+// are enabled. At CreateInstance time, the user can use the VK_EXT_validation_flags extension to pass in enum values
+// of VkValidationCheckEXT that will selectively disable checks.
+// The VK_EXT_validation_features extension can also be used with the VkValidationFeaturesEXT structure to set
+// disables in the CHECK_DISABLED struct and/or enables in the CHECK_ENABLED struct.
+struct CHECK_DISABLED {
```
Comment (tobine) [inline scripts/layer_chassis_generator.py]: Could use bit-field to aggressively pack this struct and make SetAll() function a bitwise & (false) or | (true) operation. Bits all currently fit in a DWord.

Comment (mark-lunarg) [reply scripts/layer_chassis_generator.py]: YES!  We needed to add Set helper functions, but it's pointless as things now stand.  Today I added an item to my List Of Ugly Things I Came Across to revamp the enables/disables.

**Comment by mark-lunarg** [comment]
Rebased onto Locke's changes, along with a few changes for Tony.

**Pull request merged**